### PR TITLE
fix: fix apply/validate blocks benchmarks

### PR DESCRIPTION
**Pull request by Erigara**
## Description

Fix number of issues with block benchmarks:
- correct block signing
- generating accounts correctly
- correctly get accounts by domain id
- remove account permissions on account removal
- update storage crate to fix issues with range iterator and removal of accounts 

## Benchmark results

After making account shallow we have the following result:

```plain
apply_blocks/apply_blocks
    time:   [18.491 ms 19.988 ms 21.691 ms]
    change: [+7.2201% +19.955% +34.299%] (p = 0.01 < 0.10)
    Performance has regressed.

validate_blocks/validate_blocks
    time:   [11.090 s 11.425 s 11.751 s]
    change: [-32.585% -29.986% -27.411%] (p = 0.00 < 0.10)
    Performance has improved.
```

### CODE REVIEW

File: core/src/smartcontracts/isi/domain.rs [hunk 1]
```diff
@@ -102,6 +102,11 @@ pub mod isi {
                         .expect("should succeed")
                 });
 
+            state_transaction
+                .world
+                .account_permissions
+                .remove(account_id.clone());
```
Comment (s8sato) [inline core/src/smartcontracts/isi/domain.rs:108]: I think this is a good catch, and why not the same for `world.account_roles` ?

File: core/benches/blocks/validate_blocks_oneshot.rs [hunk 2]
```diff
@@ -16,7 +18,12 @@ fn main() {
         .expect("Failed building the Runtime");
     {
         let _guard = rt.enter();
-        iroha_logger::test_logger();
+        let config = ConfigReader::new()
+            .with_env(std_env)
+            .read_and_complete()
+            .expect("Failed to load config");
+        let config = InitConfig::new(config, true);
+        let _ = iroha_logger::init_global(config).expect("Failed to initialize logger");
```
Comment (mversic) [inline core/benches/blocks/validate_blocks_oneshot.rs:26]: is this really how we construct config?

Comment (mversic) [reply core/benches/blocks/validate_blocks_oneshot.rs:26]: this is not representative of how clients will do it?

**Review by mversic — APPROVED** [review]

File: core/benches/blocks/validate_blocks_oneshot.rs (hunk 2 above)
Comment (Erigara) [reply core/benches/blocks/validate_blocks_oneshot.rs:26]: You can check how config is constructed in `read_config_and_genesis`.

Here i just wanted to use some default values and override them with env if necessary (not to spam log messages).

File: core/src/state.rs [hunk 3]
```diff
@@ -2015,15 +2004,41 @@ mod tests {
...
+
+        let domain_id = "kingdom".parse().unwrap();
+        let view = map.view();
+        let range = view
+            .range(AccountByDomainBounds::new(&domain_id))
+            .collect::<Vec<_>>();
+        assert_eq!(range.len(), 0);
+        for (account, ()) in range {
+            assert_eq!(&account.domain, &domain_id);
+        }
```
Comment (s8sato) [inline core/src/state.rs:2042]: I think this iterates over an empty vec and asserts nothing

File: core/benches/blocks/apply_blocks.rs [hunk 4]
```diff
@@ -20,17 +23,22 @@ impl StateApplyBlocks {
     /// - Failed to generate [`KeyPair`]
     /// - Failed to create instructions for block
     pub fn setup(rt: &tokio::runtime::Handle) -> Self {
-        let domains = 100;
-        let accounts_per_domain = 1000;
-        let assets_per_domain = 1000;
+        let domains = 10;
+        let accounts_per_domain = 100;
+        let assets_per_domain = 100;
```
[the remark this answers is not in the document]
Comment (Erigara) [reply core/benches/blocks/apply_blocks.rs:28]: it just took too much time...

File: core/src/state.rs (hunk 3 above)
Comment (Erigara) [reply core/src/state.rs:2042]: oh, right, it's leftover from my testing

File: core/src/smartcontracts/isi/domain.rs (hunk 1 above)
Comment (Erigara) [reply core/src/smartcontracts/isi/domain.rs:108]: good point

File: core/src/state.rs (hunk 3 above)
Comment (Erigara) [reply core/src/state.rs:2061]: fixed it to just count + added extra check

**Review by s8sato — APPROVED** [review]

**Review by mversic — APPROVED** [review]

**Pull request merged**